libs/wingsdk/src/fs/fs.ts

```diff
@@ -0,0 +1,279 @@
+import * as fs from "fs";
+import * as path from "path";
+import * as yaml from "yaml";
+import { InflightClient } from "../core";
+import { Json } from "../std";
+
+/**
+ * Convert the path to Windows format if running on Windows.
+ * @param p the path to convert
+ * @returns normalized path
+ */
+function localePath(p: string): string {
+  if (process.platform !== `win32`) {
+    return p;
+  }
+  return p.split(path.sep).join(path.win32.sep)
+}
+
+/**
+ * Custom settings for creating directory
+ */
+export interface MkdirOptions {
+  /**
+   * Indicates whether parent folders should be created.
+   * If a folder was created, the path to the first created folder will be returned.
+   * @default false
+   */
+  recursive?: boolean | undefined;
```

Comment:
The `?` implies `| undefined`:

```suggestion
  recursive?: boolean;
```